alchemiscale/base/client.py

```diff
@@ -238,6 +414,35 @@ def _post_resource(self, resource, data):
 
         return resp.json()
 
+    @_retry_async
+    @_use_token_async
+    async def _post_resource_async(self, resource, data):
+        url = urljoin(self.api_url, resource)
+        jsondata = json.dumps(data, cls=JSON_HANDLER.encoder)
+        try:
+            resp = await self._session.post(
+                url, data=jsondata, headers=self._headers, timeout=None
+            )
+        except httpx.RequestError as e:
+            raise AlchemiscaleConnectionError(*e.args)
+
+        if not 200 <= resp.status_code < 300:
+            raise self._exception(
+                f"Status Code {resp.status_code} : {resp.reason_phrase}",
+                status_code=resp.status_code,
+            )
+
+        return resp.json()
+
+    @staticmethod
```

Comment:
Very cool.

Reply:
I think Python 3.12 will have a `itertools.batched` we can just switch to.